schemas.ts

```diff
@@ -0,0 +1,172 @@
+import _ from 'lodash';
+
+/**
+ * Gets the resource schema object for a resource
+ *
+ * @param resource
+ * @param resourceName
+ * @returns The resource schema
+ */
+const getResourceSchema = (resource: any, resourceName: string): any => {
```

Comment:
Similar reasoning from the last comment. Here, I don't think we would run into any issues since there isn't a case where the returned value has more properties added to it but this could easily be the case. Again, the better solution will be to use a class/interface.